frontend/public/components/build.tsx

```diff
@@ -61,13 +61,42 @@ export const BuildsDetails: React.SFC<BuildsDetailsProps> = ({obj: build}) => {
   </div>;
 };
 
-const envPath = ['spec', 'strategy', 'sourceStrategy'];
-const environmentComponent = (props) => <EnvironmentPage
-  obj={props.obj}
-  rawEnvData={props.obj.spec.strategy.sourceStrategy}
-  envPath={envPath}
-  readOnly={true}
-/>;
+export const getStrategyType = (strategy) => {
+  switch (strategy.type) {
+    case 'Docker':
+      return 'dockerStrategy';
+    case 'Custom':
+      return 'customStrategy';
+    case 'JenkinsPipeline':
+      return 'jenkinsPipelineStrategy';
+    case 'Source':
+      return 'sourceStrategy';
+    default:
+      return null;
+  }
+};
+
+export const getEnvPath = (props) => {
+  const strategyType = getStrategyType(props.obj.spec.strategy);
+  return strategyType ? ['spec', 'strategy', strategyType] : null;
+};
+
+const environmentComponent = (props) => {
+  const {obj} = props;
+  const envPath = getEnvPath(props);
+  if (envPath) {
+    return <EnvironmentPage
+      obj={obj}
+      rawEnvData={obj.spec.strategy[getStrategyType(obj.spec.strategy)]}
+      envPath={getEnvPath(props)}
+      readOnly={true}/>;
+  }
+  return <div className="cos-status-box">
+    <div className="text-center">{`The environment variable editor doesn't support build
+      strategy: ${obj.spec.strategy.type}.`}
```

Comment:
I think this can just be

```jsx
<div className="text-center">
  The environment variable editor doesn't support build strategy: {obj.spec.strategy.type}.
</div>
```

Reply:
This will show the wording right beneath the tabbed nav. The box gives it some better spacing. Do you dislike the spacing?

Reply:
Sorry I wasn't saying get rid of the status box div. Just that you don't need the string template.

Reply:
Oh gotcha! The linter required the text to be escaped in the html elements.

Reply:
What is the exact error? It seems like the linter should allow that.

Reply:
error  HTML entities must be escaped  react/no-unescaped-entities

Reply:
Ah it wasn't the whole line, just the apos for don't that was flagging the error. Refactored the BuildEnvironmentComponent and updated the text as well.